src/Microsoft.Azure.AppConfiguration.AspNetCore/AzureAppConfigurationRefreshMiddleware.cs

```diff
@@ -24,9 +27,20 @@ public AzureAppConfigurationRefreshMiddleware(RequestDelegate next, IConfigurati
 
         public async Task InvokeAsync(HttpContext context)
         {
-            foreach (var refresher in Refreshers)
+            if (refreshReadyTime <= DateTimeOffset.UtcNow)
             {
-                _ = refresher.TryRefreshAsync();
+                //
+                // Configuration refresh is meant to execute as an isolated background task.
+                // To prevent access of request-based resources, such as HttpContext, we suppress the execution context within the refresh operation.
+                using (AsyncFlowControl flowControl = ExecutionContext.SuppressFlow())
+                {
+                    foreach (IConfigurationRefresher refresher in Refreshers)
+                    {
+                        _ = Task.Run(() => refresher.TryRefreshAsync());
+                    }
+                }
+
+                refreshReadyTime = DateTimeOffset.UtcNow.Add(TimeSpan.FromSeconds(1));
```

Comment:
Set this before doing the work to prevent other concurrent requests from doing duplicate work. I'd also suggest raising it to 15-60s, config doesn't change very often.

Also, store UtcNow in a variable to avoid calling it twice.

Reply:
Thans for the comment, added some logic with help from @jimmyca15 that should help prevent calling the logic in the if statement more than once per second. I don't think we can raise the interval though, since our current minimum refresh interval is 1 second and we wouldn't want to change that.